File: resources/expected/ros2/datatype-examples/lax/src/datatypes_system_cpp_pkg/src/base_code/producer_producer_runner.cpp
#include "datatypes_system_cpp_pkg/user_headers/producer_producer_src.hpp"

//========================================================
// Re-running Codegen will overwrite changes to this file
//========================================================

producer_producer::producer_producer() : producer_producer_base()
{
    // Invoke initialize entry point
    initialize();

    PRINT_INFO("producer_producer infrastructure set up");
}

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto executor = rclcpp::executors::MultiThreadedExecutor();
    auto node = std::make_shared<producer_producer>();
    executor.add_node(node);
    executor.spin();
    rclcpp::shutdown();
    return 0;
}

File: resources/expected/ros2/python-datatype-examples/strict/src/datatypes_system_py_pkg/datatypes_system_py_pkg/base_code/producer_producer_runner.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from rclpy.executors import MultiThreadedExecutor
from datatypes_system_py_pkg.user_code.producer_producer_src import producer_producer
#========================================================
# Re-running Codegen will overwrite changes to this file
#========================================================
def main(args=None):
    rclpy.init(args=args)
    node = producer_producer()
    executor = MultiThreadedExecutor()
    executor.add_node(node)
    executor.spin()
    rclpy.shutdown()

if __name__ == "__main__":
    main()
